src/sparse_blas/function_table.hpp

```diff
@@ -0,0 +1,291 @@
+/*******************************************************************************
+* Copyright 2023 Codeplay Software Ltd.
+*
+* (*Licensed under the Apache License, Version 2.0 )(the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions
+* and limitations under the License.
+*
+*
+* SPDX-License-Identifier: Apache-2.0
+*******************************************************************************/
+
+#ifndef _ONEMKL_SPARSE_BLAS_FUNCTION_TABLE_HPP_
+#define _ONEMKL_SPARSE_BLAS_FUNCTION_TABLE_HPP_
+
+#include "oneapi/mkl/sparse_blas/types.hpp"
+
+typedef struct {
+    int version;
+    void (*init_matrix_handle)(sycl::queue &queue, oneapi::mkl::sparse::matrix_handle_t *p_handle);
+
+    sycl::event (*release_matrix_handle)(sycl::queue &queue,
+                                         oneapi::mkl::sparse::matrix_handle_t *p_handle,
+                                         const std::vector<sycl::event> &dependencies);
+
+    // set_csr_data
+    void (*set_csr_data_buffer_rf_i32)(sycl::queue &queue,
+                                       oneapi::mkl::sparse::matrix_handle_t handle,
+                                       std::int32_t num_rows, std::int32_t num_cols,
+                                       std::int32_t nnz, oneapi::mkl::index_base index,
```

Comment:
Maybe we could use those `INSTANTIATE`-type macros here as well to avoid subtle mistakes down the road?

Reply:
Done in https://github.com/oneapi-src/oneMKL/pull/374/commits/5341802